Source/UI/IdentityUI.swift

```diff
@@ -477,6 +488,105 @@ extension IdentityUI {
+                fallback()
+                return
+            }
+
+            self.identityManager.login(username: .email(email),
+                                       password: password,
+                                       scopes: scopes,
+                                       persistUser: true,
+                                       useSharedWebCredentials: false) { result in
+                guard case .success = result else {
+                    fatalError()
+                    return
+                }
+
+                let user = self.identityManager.currentUser
+
+                func success() {
+                    completion(.success(user: user, persistUser: true))
+                }
+
+                func cancel() {
+                    completion(.cancel)
+                }
+
+                func presentTerms() {
+                    self.identityManager.fetchTerms { result in
+                        guard case .success(let terms) = result else {
+                            cancel()
+                            return
+                        }
+
+                        let configuration = IdentityUIConfiguration(
+                            clientConfiguration: user.clientConfiguration)
+
+                        let navigationController = DismissableNavigationController {
+                            if IdentityUI.updatedTermsCoordinator != nil {
+                                user.logout()
+                                cancel()
+                            }
+                            IdentityUI.updatedTermsCoordinator = nil
+                        }
+
+                        let updatedTermsCoordinator = UpdatedTermsCoordinator(
+                            navigationController: navigationController, configuration: configuration)
+
+                        updatedTermsCoordinator.start(input: .init(currentUser: user, terms: terms)) { output in
+                            IdentityUI.updatedTermsCoordinator = nil
+
+                            navigationController.dismiss(animated: true) {
+                                switch output {
+                                case .success:
+                                    success()
+                                case .cancel:
+                                    cancel()
+                                }
+                            }
+                        }
+
+                        IdentityUI.updatedTermsCoordinator = updatedTermsCoordinator
+                        configuration.presentationHook?(navigationController)
+                        presentingViewController.present(navigationController, animated: true, completion: nil)
+                    }
+                }
+
+                user.agreements.status { result in
+                    guard case .success(let agreed) = result, agreed else {
+                        presentTerms()
+                        return
+                    }
+
+                    success()
```

Comment:
Why do we success if agreements cannot be checked?

Reply:
We're not, you're reading the code wrong. We succeed if we read the agreements _and_ they were agreed to.